flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/source/utils/TableDiscoveryUtils.java

```diff
@@ -50,54 +54,26 @@ public static List<TableId> listTables(
         // READ DATABASE NAMES
         // -------------------
         // Get the list of databases ...
-        LOG.info("Read list of available databases");
-        final List<String> databaseNames = new ArrayList<>();
-
-        jdbc.query(
-                "SHOW DATABASES",
-                rs -> {
-                    while (rs.next()) {
-                        String databaseName = rs.getString(1);
-                        if (databaseFilter.test(databaseName)) {
-                            databaseNames.add(databaseName);
-                        }
+        try (Connection connection = jdbc.connection()) {
+            DatabaseMetaData databaseMetaData = connection.getMetaData();
+            try (ResultSet tableResult = databaseMetaData.getTables(null, null, "%", TABLE_QUERY)) {
+                while (tableResult.next()) {
+                    String dbName = tableResult.getString("TABLE_CAT");
+                    if (!databaseFilter.test(dbName)) {
```

Comment:
We should skip the whole tables of this database directly.

Reply:
You can ignore this comment, reducing the number of interactions with the database is more appropriate than reducing calculations.